libraries/utilities.js

```diff
@@ -276,6 +276,27 @@ var getObjectIdFromTargetOrObjectFile = function (folderName, objectsPath) {
 };
 exports.getObjectIdFromTargetOrObjectFile = getObjectIdFromTargetOrObjectFile;
 
+var getAnchorIdFromObjectFile = function (folderName, objectsPath) {
+
+    if (folderName === 'allTargetsPlaceholder') {
+        return 'allTargetsPlaceholder000000000000';
+    }
+
+    var jsonFile = objectsPath + '/' + folderName + '/object.json';
+
+    if (fs.existsSync(jsonFile)) {
+        let thisObject = JSON.parse(fs.readFileSync(jsonFile, 'utf8'));
+        if (thisObject.hasOwnProperty('objectId')) {
+            return thisObject.objectId;
+        } else {
+            return null;
+        }
+    } else {
+        return null;
+    }
+};
+exports.getAnchorIdFromObjectFile = getAnchorIdFromObjectFile;
```

Comment:
This is a good solution but I really want to solve this at the root of the problem – the object.json file at this path shouldn't exist (should always be inside the .identity folder). I actually had no idea that that bug existed until you noticed it.